diff --git a/libarchive/filter_fork_posix.c b/libarchive/filter_fork_posix.c
index 7c48519..075d73b 100644
--- a/libarchive/filter_fork_posix.c
+++ b/libarchive/filter_fork_posix.c
@@ -26,6 +26,10 @@
 
 #include "archive_platform.h"
 
+#if defined(__APPLE__)
+	#include <TargetConditionals.h>
+#endif
+
 /* This capability is only available on POSIX systems. */
 #if defined(HAVE_PIPE) && defined(HAVE_FCNTL) && \
     (defined(HAVE_FORK) || defined(HAVE_VFORK) || defined(HAVE_POSIX_SPAWNP))
@@ -74,6 +78,9 @@ int
 __archive_create_child(const char *cmd, int *child_stdin, int *child_stdout,
 		pid_t *out_child)
 {
+#if defined(__APPLE__) && !TARGET_OS_OSX && !TARGET_OS_MACCATALYST
+	return ARCHIVE_FAILED;
+#else
 	pid_t child = -1;
 	int stdin_pipe[2], stdout_pipe[2], tmp;
 
@@ -224,6 +231,7 @@ stdin_opened:
 state_allocated:
 	__archive_cmdline_free(cmdline);
 	return ARCHIVE_FAILED;
+#endif /* defined(__APPLE__) && !TARGET_OS_OSX && !TARGET_OS_MACCATALYST */
 }
 
 void
